core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -586,6 +612,16 @@ protected void writeSnapshotList(List<SnapshotId> snapshots) throws IOException
      * @throws IOException I/O errors
      */
     protected List<SnapshotId> readSnapshotList() throws IOException {
+        try {
+            if (snapshotsBlobContainer.blobExists(SNAPSHOTS_FILE) == false) {
```

Comment:
Just to follow up, you are correct, we do not need this check here, because as it stands, `readSnapshotList` is never invoked before it is written.  In subsequent work, we will probably have to re-introduce this check, but I'm removing it from this PR for now as it is not necessary for it.